src/main/java/com/google/sps/servlets/Post.java

```diff
@@ -0,0 +1,160 @@
+    int endMinute = 0;
+    String location = "";
+    Double lat = 0.0;
+    Double lng = 0.0;
+    String numberOfPeopleItFeeds = "";
+    String typeOfFood = "";
+    String description = "";
+    String collegeId = "";
+    int timeSort = 0;
+
+    public Post() {
+        // All variables are already initialized.
+    }
+
+    /* Fill in the important Post details from the POST request. */
+    public void requestToPost(HttpServletRequest request, String collegeId) {
+        organizationName = request.getParameter("organizationName");
+        month = Integer.parseInt(request.getParameter("month")) - 1; // Months are indexed at 0.
+        day = Integer.parseInt(request.getParameter("day"));
+        year = Integer.parseInt(request.getParameter("year"));
+        startHour = Integer.parseInt(request.getParameter("startHour"));
+        startMinute = Integer.parseInt(request.getParameter("startMinute"));
+        endHour = Integer.parseInt(request.getParameter("endHour"));
+        endMinute = Integer.parseInt(request.getParameter("endMinute"));
+        location = request.getParameter("location");
+        lat = Double.parseDouble(request.getParameter("lat"));
+        lng = Double.parseDouble(request.getParameter("lng"));
+        numberOfPeopleItFeeds = request.getParameter("numberOfPeopleItFeeds");
+        typeOfFood = request.getParameter("typeOfFood");
+        description = request.getParameter("description");
+        this.collegeId = collegeId;
+
+        // Adjust the start and end hour based on whether the hour is AM or PM.
+        String startAMorPM = request.getParameter("startAMorPM");
+        if (startAMorPM.equals("pm")) {
+            startHour += 12;
+        }
+        String endAMorPM = request.getParameter("endAMorPM");
+        if (endAMorPM.equals("pm")) {
+            endHour += 12;
+        }
+
+        // Translate the start time into minutes to allow for sorting.
+        timeSort = startHour * 60 + startMinute;
+    }
+
+    /* Translate the entities from the Datastore query to Post objects and return in an array. */
+    public static ArrayList<Post> queryToPosts(PreparedQuery queryResult, DatastoreService datastore) {
+        ArrayList<Post> currentPosts = new ArrayList<Post>();
+
+        // Create a calendar based off the current time zone.
+        // TODO: Update Time Zone based off student and university location, instead of "America/Los_Angeles".
+        Calendar nowTime = Calendar.getInstance(TimeZone.getTimeZone("America/Los_Angeles"));
+
+        for (Entity entity: queryResult.asIterable()) {
+
+            // Create a calendar based off the post timing.
+            Calendar postTime = Calendar.getInstance(TimeZone.getTimeZone("America/Los_Angeles"));
+            int postMonth = ((Long) entity.getProperty("month")).intValue();
+            int postDay = ((Long) entity.getProperty("day")).intValue();
+            int postYear = 2000 + ((Long) entity.getProperty("year")).intValue();
```

Comment:
Do we have bounds for years/edge case checking for irregular dates? No worries if you don't, can be an issue/to-do for later!

Reply:
For the number input, I have it so that it accepts only 20-30. But the user can definitely get around this, so I'll make an issue around it!

Reply:
And add tests as well. Can be done in a subsequent PR.